### ISSUE TITLE
Support for MKS TFT32_L V4.0

### ISSUE THREAD

**Issue by paloky** [darkspr1te/MKSTFT_Marlin_Touch#16]
Hi darkspr1te.
Have you support for the  MKS TFT32_L V4.0 ??

Thank a lot.

**Comment by darkspr1te** [comment]
Yes, Btt have accepted my pull request and it's now part of official BTT build , found here https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware 
Select MKS_32_V1_4 as your model to build 
darkspr1te

**Comment by luc-github** [comment]
@darkspr1te I have an MKS TFT32_L V4.0 and flashed with your porting using SD methode in May and it is working well - thank you so much

Now  I woulld like to update with latest code, I have tried to use latest BTT git binary and compiled also latest BTT git and I cannot update it anymore. I have also tried your git. 
I used native name MKS_32_V1_4.26.bin and tried mkstft32.bin on SD  - the file seems ignored and so FW is not updated
I have also tried to put it in MKS directory and it is still ignored
bmp and font are always seen and updated but not FW itself

Do I miss something ?  Did you had such behavior before ? I am guessing it is some file name issue but cannot figure out the solution

**Comment by Uz45** [comment]
Your need use this name mkstft28.bin :)

**Comment by luc-github** [comment]
orz... @Uz45 thank you ^_^

**Comment by paloky** [comment]
Hi @Uz45.
So I have to take the file "MKS_32_V1_4.26.x.bin" of the folder "Copy to SD Card root directory to update", and rename it in the SD as "mkstft28.bin" ?
Apart from using the "MKS" folder with the BMP and the FONTS, right?

Is that so?
Then I will try to see if it works for me.

Thanks.

**Comment by Uz45** [comment]
> Hi @Uz45.
> So I have to take the file "BIGTREE_TFT28_V3.0.25.1.bin" of the forlder "Copy to SD Card root directory to update", and rename it in the SD as "mkstft28.bin" ?
> Apart from using the "MKS" folder with the BMP and the FONTS, right?
> 
> Is that so?
> Then I will try to see if it works for me.
> 
> Thanks.

Hi @paloky.
You need the file "MKS_32_V1_4.26.x.bin". Rename it to "mkstft28.bin" and put it on SD. And put the MKS folder on the SD. I recommend creating an empty text file on the SD card "reset.txt" to clear the eprom. This will help you avoid many problems.

**Comment by paloky** [comment]
Hi @Uz45.
The Boot .bin works, but next, when upload the BMP and FONTS, the screen show a rectangles.
Finaly, show me the touch calibration, and next reboot and show the next image.

[https://drive.google.com/drive/folders/1yaZQWP-PaB3D15ShC05aJ5nhh63oMJz4?usp=sharing](url)

**Comment by Uz45** [comment]
> Hi @Uz45.
> The Boot .bin works, but next, when upload the BMP and FONTS, the screen show a rectangles.
> Finaly, show me the touch calibration, and next reboot and show the next image.
> 
> [https://drive.google.com/drive/folders/1yaZQWP-PaB3D15ShC05aJ5nhh63oMJz4?usp=sharing](url)

@paloky.
This happens when fonts and images are not loaded. I had this problem when I copied the font and bmp folders from the MKS folder to the SD. Copy the entire "MKS" folder to the SD root.

**Comment by paloky** [comment]
Hi @Uz45.
Right !!!..  If I set the MKS folder, works !!!.

But now. Show me the next Error after calibration and reboot.

```
Found Updated data:

Fonts: Update requiered
Icons: Update required

Insert the SD card with the required files and press the reset button to update.
```


I try to upload the "Unified Menu" and "Clasic Menu" and with both I get this result.

Any Idea about this issu?

**Comment by paloky** [comment]
Hi. I solved the Issue.
Finaly I used the "Unified Menu", but I need to use the "TFT28" folder and rename to "MKS" in SD card.
Now I don't find 2 bitmaps:  (I create my own BMP later..).

-    S_SLAVE.BMP
-    S_LOAD.BMP

And Error in configuration file:

-    mesh_min_color
-    mesh_max_color


But now, The TFT Works fine !!!
Thank a lot.

**Comment by Uz45** [comment]
Hi@paloky.

The versions may not match. The bin file is rarely updated and it happens that the firmware version is old, but the images are already for the new version.
